File: data/compilers_comparation/src/examples/second_example/src/func57.c
#include "second_example.h" 
array_t* func57(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array38;
   if (pCounter > 0) {
      array38 = vars->data[--pCounter];
      array38->refC++;
      DEBUG_COPY(array38->id);
   } else {
      array38 = (array_t*)malloc(sizeof(array_t));
      array38->size = 306;
      array38->refC = 1;
      array38->id = 38;
      array38->data = (unsigned int*)malloc(array38->size*sizeof(unsigned int));
      memset(array38->data, 0, array38->size*sizeof(unsigned int));
      DEBUG_NEW(array38->id);
   }
   return array38;
}

File: data/multilang_comparison/program_nim/src/func57.nim
proc func57*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array59: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array59 = vars.data[pCounter]
      inc array59.refC
      debugCopy(array59.id)
   else:
      array59 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array59.size = 764
      array59.refC = 1
      array59.id = 59
      array59.data = cast[ptr UncheckedArray[uint32]](alloc(array59.size * sizeof(uint32)))
      zeroMem(array59.data, array59.size * sizeof(uint32))
      debugNew(array59.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array59
   var array60: ptr ArrayT = func59(addr params0, loopsFactor)
   debugReturn(array60.id)
   dealloc(params0.data)
   dec array60.refC
   if array60.refC == 0:
      dealloc(array60.data)
      dealloc(array60)
      debugFree(array60.id)
   return array59

